python/src/synapse/ml/dl/DeepVisionClassifier.py

```diff
@@ -0,0 +1,320 @@
+# Copyright (C) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See LICENSE in project root for information.
+
+import sys
+
+import torchvision.transforms as transforms
+from horovod.spark.common.backend import SparkBackend
+from PIL import Image
+from pyspark import keyword_only
+from pyspark.context import SparkContext
+from pyspark.ml.param.shared import Param, Params
+from synapse.ml.dl.LitDeepVisionModel import LitDeepVisionModel
+from synapse.ml.dl.DeepVisionModel import DeepVisionModel
+from horovod.spark.lightning import TorchEstimator
+from pytorch_lightning.utilities import _module_available
+
+_HOROVOD_AVAILABLE = _module_available("horovod")
+if _HOROVOD_AVAILABLE:
+    import horovod
+
+    _HOROVOD_EQUAL_0_25_0 = horovod.__version__ == "0.25.0"
+    if not _HOROVOD_EQUAL_0_25_0:
+        raise RuntimeError(
+            "horovod should be of version 0.25.0, found: {}".format(horovod.__version__)
+        )
+else:
+    raise ModuleNotFoundError("module not found: horovod")
+
+
+class DeepVisionClassifier(TorchEstimator):
+
+    backbone = Param(
+        Params._dummy(), "backbone", "backbone of the deep vision classifier"
+    )
+
+    additional_layers_to_train = Param(
+        Params._dummy(),
+        "additional_layers_to_train",
+        "number of last layers to fine tune for the model, should be between 0 and 3",
+    )
+
+    num_classes = Param(Params._dummy(), "num_classes", "number of target classes")
+
+    loss_name = Param(
+        Params._dummy(),
+        "loss_name",
+        "string representation of torch.nn.functional loss function for the underlying pytorch_lightning model, e.g. binary_cross_entropy",
+    )
+
+    optimizer_name = Param(
+        Params._dummy(),
+        "optimizer_name",
+        "string representation of optimizer function for the underlying pytorch_lightning model",
+    )
+
+    dropout_aux = Param(
+        Params._dummy(),
+        "dropout_aux",
+        "numeric value that's applied to googlenet InceptionAux module's dropout layer only: probability of an element to be zeroed",
+    )
+
+    transform_fn = Param(
```

Comment:
Is this expecting a pytorch transformation, or a func that transforms rows? The former would be a much nicer API for this

Reply:
Yes this is expecting a pytorch transformation, the func that transforms rows is TorchEstimator's transformation_fn parameter.